man/faq/user/faq-compatibility-types.Rmd

```diff
@@ -0,0 +1,62 @@
+
+```{r, child = "../setup.Rmd", include = FALSE}
+```
+
+```{r, include = FALSE}
+stopifnot(rlang::is_installed("dplyr"))
+```
+
+Two vectors are __compatible__ when you can safely:
+
+- Combine them into one larger vector.
+- Assign values from one of the vector in the other vector.
+
+Examples of compatible types are integer and double vectors. On the other hand, integer and character vectors are not compatible.
```

Comment:
Integer and doubles are interesting because integer->double is always compatible, but you can't always assign a double into an integer vector. Not sure if this is worth mentioning, but safe assignment was one of your two defining principles for compatibility so I figured I'd bring it up.

Reply:
Update: I see now that you mention the integer/double issue below in the conversion section. Is the compatibility principle above regarding assignment sufficiently met when you can always assign in at least one of the two directions? (i.e. we can always assign integer->double, but not always double->integer, but that is enough for them to be compatible)

Reply:
Yes I framed "compatible" to mean "have a common type". Do you think that makes sense? The lossy cast errors are not compatible type errors:

```r
vec_assign(1:3, 2, 1.5)
#> Error: Can't convert from <double> to <integer> due to loss of precision.
#> * Locations: 1

class(last_error())
#> [1] "vctrs_error_cast_lossy" "vctrs_error"            "rlang_error"
#> [4] "error"                  "condition"
```

Reply:
Yea, after reading the whole article I think it makes sense as is. If I was a new user, I think this would help a lot.